### PR TITLE
Scary stories

### PR DESCRIPTION
**Pull request by aldoram5**
Now Monika can tell you some scary stories ~~and try to scare you the same way ddlc did~~.
This one adds 12 new scary stories that Monika can tell the player.

## Changes
- There's going to be a new topic `Can you tell me a horror story?` that's going to trigger the flow for the scary stories. First time she'll jump straight into the first story, consecutive times she'll ask the player to pick from a menu the story they want to hear.
- There's a new category `TYPE_SCARY` for stories.
- There's a new script that contains some of the definitions for the effects used on some of the stories, it's called `special-effects.rpy`
- There's no scary stories for sensitive mode since these include content `not suitable for children\nor those who are easily disturbed.`
- Adds a function to `mas_stories` store that unlocks every story for easy testing. Function is `_unlock_everything()`. 

## Testing
- Scary stories unlock the same way that normal stories do, one each session.
- Set `mas_full_scares` to True, to see the scares that some stories have.
- Check that the topic for the stories isn't available on sensitive mode.
- Check that the stories correctly restore the game state after the story ends.
- If you're no longer testing the unlock flow you can execute on the console `mas_stories._unlock_everything()` to unlock every story.

### Special Thanks
- @kaido1224 for the stories and general suggestions.
- @sebastianN01 for the art.
- @multimokia for the scary soundtrack.

### CODE REVIEW

**Comment by ThePotatoGuy** [comment]
because of the size of this change, I'm adding @multimokia as an additional tester. multi when you have finished testing and everything works fine, approve the pull request.

**Comment by kaido1224** [comment]
Happy to be of service.

File: Monika After Story/game/script-stories.rpy [hunk 1]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+label mas_scary_story_setup:
+    show monika 1dsc
+    $ mas_temp_r_flag = mas_is_raining
+    $ scene_change = True
+    $ mas_is_raining = True
+    pause 1.0
+    $ mas_temp_m_flag = morning_flag
+    call spaceroom(start_bg="monika_gloomy_room")
+    $ morning_flag = True
+    #stop music fadeout 1.0
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: can you fade out the music and play the ambience track multi uploaded on the discord?

File: Monika After Story/game/script-stories.rpy [hunk 2]
```diff
@@ -10,11 +10,16 @@
 # dict of tples containing the stories event data
 default persistent._mas_story_database = dict()
 default mas_can_unlock_story = False
+default mas_can_unlock_scary_story = False
+default mas_full_scares = False
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: is this supposed to be the whether or not like they horrors/spooks?

if so, then mark it with a todo/comment so we can change this when the o31 stuff gets in.

File: Monika After Story/game/script-stories.rpy [hunk 3]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+    $ mas_temp_m_flag = morning_flag
+    call spaceroom(start_bg="monika_gloomy_room")
+    $ morning_flag = True
+    #stop music fadeout 1.0
+    play background audio.rain fadein 1.0 loop
+    show vignette zorder 13
+#    $ songs.current_track = songs.FP_NO_SONG
+#    $ songs.selected_track = songs.FP_NO_SONG
+
+    $ HKBHideButtons()
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: Can you also disable escape menu? and the hotkeys?

Might as well raise the core ui shields tbh.

File: Monika After Story/game/script-stories.rpy [hunk 4]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+    m "The salesman then suddenly appeared to the hunter while he was grieving his loss, revealing that he was in fact the devil."
+    m 4esd "He gives the hunter a chance to redeem for his mistake."
+    m 4esb "If the hunter remained faithful to his beloved who was slain for the remainder of his life, he would be reunited with her after death."
+    m 1eud "The hunter vowed to remain true to his beloved and remained faithful for a time, but soon fell in love with another girl and married her."
+    m "One year to the day after the fatal incident, as the hunter rode through the forest chasing some game, he comes across the spot where he slayed his beloved."
+    m 2wuo "To his horror,{w=1.0} her corpse, which was buried elsewhere, was standing in the same spot she was slain."
+    m "She approached the hunter, scorning him for being unfaithful and vowing revenge for slaying her."
+    m "The hunter rides away in a panic."
+    m 1esd "After a short way, he looks behind him to see if she was following him any longer."
+    m 1wuo "To his horror, not only had he gained no distance, she had gained on him significantly."
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: `To his horror `twice in 5 lines? replace one of them with something different.

File: Monika After Story/game/script-stories.rpy [hunk 5]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+            "Did I scare you, " + player + "?",
+            "How was it?"
+        ]
+        story_end_quip=renpy.random.choice(story_end_quips)
+
+    m 3eua "[story_end_quip]"
+    if not mas_temp_r_flag:
+        stop background fadeout 1.0
+    show monika 1dsc
+    $ scene_change = True
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: on o31, we will have this sort of scary theme persist beyond stories, so can you add a todo here so o31 can adjust this later

File: Monika After Story/game/script-stories.rpy [hunk 6]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+            "am I pretty?"
+            "Yes":
+                hide screen mas_background_timed_jump
+                jump mas_scary_story_kuchisake_onna.clean
+            "No":
+                jump mas_scary_story_kuchisake_onna.no
+    else:
+        jump mas_scary_story_kuchisake_onna.end
+
+    label .no:
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: so we had issues with local labels in the chess cheating dialogue flow. My understanding was that the label needed to be reached first before we could reference it in jumps/calls. Is that not the case?

Note that this kind of issue only surfaces when using a release version, not via renpy sdk.

File: Monika After Story/game/script-stories.rpy [hunk 7]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+    m 1tsb "Not too long after they said this, there came a knock at their door."
+    m 3eub "There in the doorway stood four beautiful women."
+    m "The women, having become lost in the wilderness, asked if they may join the men in their shelter for the night."
+    m 1tku "The men, silently congratulating themselves on their good fortunes, invited the women in."
+    m 1esa "After a while of enjoying each other’s company, the women expressed a wish to dance."
+    m 1tku "The men wasted no time coupling off with each of the maidens."
+    m 1eub "As they are dancing, one of the men notices that the other couples are dancing rather erratically."
+    m 1wuo "Then, to his horror, he realizes that the other men have blood pouring from their necks onto their shirts."
+    m 3wuo "In a blind panic, the man abandoned his partner and bolted out the door, which he was fortunate enough to be close to, before he could share the fate of his friends."
+    m 3wud "The man ran into the forest and ended up hiding amongst the horses he and his friends had ridden during that day’s hunt."
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]:
```suggestion
    m 3wud "The man ran into the forest and hid amongst the horses he and his friends had ridden during that day’s hunt."
```

File: Monika After Story/game/script-stories.rpy [hunk 8]
```diff
@@ -126,36 +151,76 @@ label mas_story_begin:
     return
 
 label mas_story_unlock_random:
+   call mas_story_unlock_random_cat()
+   return
+
+label mas_scary_story_unlock_random:
+   call mas_story_unlock_random_cat(scary=True)
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: could you add a dev function that can unlock and see all the stories?

calling this label works but it gives me them randomly, which is a minor annoyance when trying to test/review some of the action code.

File: Monika After Story/game/script-stories.rpy [hunk 9]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+    return
+
+init 5 python:
+    addEvent(Event(persistent._mas_story_database,eventlabel="mas_scary_story_serial_killer",
+    category=[store.mas_stories.TYPE_SCARY], prompt="The serial killer",unlocked=False),
+    eventdb=store.mas_stories.story_database)
+
+label mas_scary_story_serial_killer:
+    call mas_scary_story_setup
+    m 4esb "A young couple park at a cemetery one night for some undisturbed lovemaking."
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]:
```suggestion
    m 4esb "A young couple parked their car at a cemetery one night for some undisturbed lovemaking."
```

File: Monika After Story/game/script-stories.rpy [hunk 10]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+    m 3eud "The police officer instructs the girl to get out of the car, walk towards him and not look back."
+    m "She does as he instructs her to. Not too long after, more police cars arrive."
+    m "Curiosity gets the better of the girl and she looks behind her."
+    m 4wuo "She is horrified to see the body of her boyfriend hanging head down from a tree, his throat slit from ear to ear."
+    # chance of 1 in 8
+    if renpy.random.randint(1,8) == 1 or mas_full_scares:
+        show y_sticker hopg zorder 17:
+            pos(600,425)
+            alpha 1.0
+            linear 1.0 alpha 0
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: this linear is too short. should try to make this so the sound ends as yuri sticker disappears

File: Monika After Story/game/script-stories.rpy [hunk 11]
```diff
@@ -348,11 +412,556 @@ label mas_story_immortal_love:
...
+    if renpy.random.randint(1,2) == 1 or mas_full_scares:
+        play sound "sfx/crack.ogg"
+    m 3wuo "He looked up and in the light of the lamp, he saw the woman rise, now bearing fangs and fingernails that looked like claws, advancing towards them."
+    m "She bent down and bit each of the sleeping men. The fourth man, at the last second, pulled up a pillow in front of his neck."
+    m 1eud "The woman bit the pillow and apparently not realizing she hadn’t bit the last man, returned to her original resting spot."
+    m 3eud "The man kicked his companions, but none of them moved. The man decided to take his chances and make a run for it."
+    m 3wuo "As soon as his feet touched the ground, however, he heard another creak."
+    m "Realizing that the woman was again rising from her spot, he opened the door and ran as fast as he could."
+    # need opinions on this one since it's for storytelling purposes
+    show layer master at heartbeat2(1)
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: should also add the heartbeat sound here too

File: Monika After Story/game/script-stories.rpy (hunk 2 above)
Comment (aldoram5) [reply Monika After Story/game/script-stories.rpy]: Nope it's just for easy testing

File: Monika After Story/game/script-stories.rpy (hunk 3 above)
Comment (aldoram5) [reply Monika After Story/game/script-stories.rpy]: yeah I was only disabling the others but makes sense to disable everything.

File: Monika After Story/game/script-stories.rpy (hunk 5 above)
Comment (aldoram5) [reply Monika After Story/game/script-stories.rpy]: sure

File: Monika After Story/game/script-stories.rpy (hunk 6 above)
Comment (aldoram5) [reply Monika After Story/game/script-stories.rpy]: I'll test on a release version and change accordingly to what my tests show

File: Monika After Story/game/script-stories.rpy (hunk 8 above)
Comment (aldoram5) [reply Monika After Story/game/script-stories.rpy]: sure it'll be used to unlock any locked story.

File: Monika After Story/game/script-stories.rpy (hunk 10 above)
Comment (aldoram5) [reply Monika After Story/game/script-stories.rpy]: sure

File: Monika After Story/game/script-stories.rpy (hunk 11 above)
Comment (aldoram5) [reply Monika After Story/game/script-stories.rpy]: 📈

File: Monika After Story/game/script-stories.rpy [hunk 12]
```diff
@@ -348,11 +412,555 @@ label mas_story_immortal_love:
...
+    return
+
+label mas_scary_story_cleanup:
+    python:
+        story_end_quips = [
+            "Scared, " + player + "?",
+            "Did I scare you, " + player + "?",
+            "How was it?"
+        ]
+        story_end_quip=renpy.random.choice(story_end_quips)
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-stories.rpy]: can you just apply `renpy.substitute` to the selected quip, so the quips can use the regular renpy string formatting codes?


https://www.renpy.org/doc/html/other.html#renpy.substitute

File: Monika After Story/game/script-stories.rpy [hunk 13]
```diff
@@ -348,11 +421,573 @@ label mas_story_immortal_love:
...
+#Stopping point - Edit done. Review exps
+label mas_scary_story_kuchisake_onna:
+    call mas_scary_story_setup
+    m 3eud "There once was a beautiful woman, whom was the wife of a samurai."
+    m 3eub "She was as incredibly beautiful as she was vain, welcoming the attention of any man prepared to offer it to her."
+    m 1tsu "And often, would ask men to appraise her appearance."
+    m 1esd "The woman was prone to cheat on her husband multiple times and was soon found out about her affairs."
+    m 1efd "When he confronted her, he was beyond infuriated as she was damaging their status as nobles, humiliating him."
+    m "He then brutally punished her by cutting her mouth from ear to ear, disfiguring her delicate beauty."
+    m "'Who will think as beauitful now?' were his haunting words to her."
```
Comment (multimokia) [inline Monika After Story/game/script-stories.rpy]:
```suggestion
    m "'Who will think as beautiful now?' were his haunting words to her."
```

File: Monika After Story/game/script-stories.rpy [hunk 14]
```diff
@@ -348,11 +421,573 @@ label mas_story_immortal_love:
...
+    m 2dsd "Shortly after, the woman died, unable to live further after she was tarnished and treated like a freak by everyone around he."
```
Comment (multimokia) [inline Monika After Story/game/script-stories.rpy]:
```suggestion
    m 2dsd "Shortly after, the woman died, unable to live further after she was tarnished and treated like a freak by everyone around her."
```

**Pull request merged**